### PR TITLE
import instance wrapper

### PR DESCRIPTION
**Pull request by ksobon**
### Purpose

This adds wrapper to `ImportInstance` class so that its not returned as `UnknownElement`. It was raised here: https://github.com/DynamoDS/DynamoRevit/issues/1410

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@ikeough @mjkkirschner @aparajit-pratap 

### FYIs

- None

### CODE REVIEW

**Comment by aparajit-pratap** [comment]
@ksobon could you attach an example Revit file and DYN as a potential test case for this fix? We can then have that included in our test suite. Thanks!

**Comment by ksobon** [comment]
@aparajit-pratap This is really a minor change where we were missing a `FromExisting` implementation on the `ImportInstance` class. That's the only thing that I added here. There are no other changes and whatever tests you were performing previously should not be affected. 

Do you need a Revit + Dyn file to be placed in the usual repo location with the rest of the test models? I can make a small model, but I am not sure its necessary.

**Comment by kronz** [comment]
tracking review process in https://jira.autodesk.com/browse/DYN-483

File: src/Libraries/RevitNodes/Elements/ImportInstance.cs [hunk 1]
```diff
@@ -27,7 +27,22 @@ public override Autodesk.Revit.DB.Element InternalElement
...
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Set the internal Element, ElementId, and UniqueId
+        /// </summary>
+        /// <param name="element"></param>
+        private void InternalSetElement(Autodesk.Revit.DB.ImportInstance element)
```
Comment (aparajit-pratap) [inline src/Libraries/RevitNodes/Elements/ImportInstance.cs]: @ksobon I don't think this method is necessary. We already have `InternalSetImportInstance`

File: src/Libraries/RevitNodes/Elements/ImportInstance.cs [hunk 2]
```diff
@@ -82,6 +106,15 @@ private void InitImportInstance(string satPath, XYZ translation = null)
             ElementBinder.SetElementForTrace(importInstance);
         }
 
+        /// <summary>
+        /// Initilize an ImportInstance from existing element
+        /// </summary>
+        /// <param name="instance"></param>
+        private void InitImportInstance(Autodesk.Revit.DB.ImportInstance instance)
```
Comment (aparajit-pratap) [inline src/Libraries/RevitNodes/Elements/ImportInstance.cs]: I think we can get rid of this as well and just use `InternalSetImportInstance`

**Comment by aparajit-pratap** [comment]
@ksobon some minor comments. Other than that it is good to go.

**Comment by aparajit-pratap** [comment]
@ksobon I'm not sure if it's just me but I tested your changes and am getting the following error with a simple rvt file. Can you reproduce this?
image

**Comment by aparajit-pratap** [comment]
Ok the issue was lurking from earlier - `InternalSetImportInstance` was incorrect. It should be changed to:
```
private void InternalSetImportInstance(Autodesk.Revit.DB.ImportInstance ele)
        {
            this.InternalImportInstance = ele;
            this.InternalElementId = ele.Id;
            this.InternalUniqueId = ele.UniqueId;
        }
```

**Comment by aparajit-pratap** [comment]
@ksobon please take a look at https://github.com/DynamoDS/DynamoRevit/pull/1575 for the test case. Thanks for the fix!

**Comment by ksobon** [comment]
@aparajit-pratap oh wow! I didn't see that coming. My bad. Why does the order matter here? It's kind of disturbing that we would remember something like this for the future and there was no "obvious" reason for this specific order.

**Comment by aparajit-pratap** [comment]
@ksobon that seems strange to me too but I don't have an answer right now. Will have to investigate this in more detail to get more clarity but for now, I'm going ahead and closing this PR and merging #1575 which has the test case. Thanks again!

**Pull request closed without merging**